### PR TITLE
Add Additional Logging Options

### PR DESCRIPTION
**Pull request by thefringeninja**
Added: New options to control logging: --log-console-format, --log-file-size, --log-file-interval, --log-file-retention-count, and --disable-log-file
Fixed: Incorrect configuration file selection on linux

### CODE REVIEW

File: src/EventStore.Common/Log/EventStoreLoggerConfiguration.cs [hunk 1]
```diff
@@ -43,7 +44,9 @@ public class EventStoreLoggerConfiguration {
 			};
 		}
 
-		public static void Initialize(string logsDirectory, string componentName, string logConfig = "logconfig.json") {
+		public static void Initialize(string logsDirectory, string componentName, LogConsoleFormat logConsoleFormat,
+			int logFileSize, RollingInterval logFileInterval, int logFileRetentionCount, bool disableLogFile,
+			string logConfig = "logconfig.json") {
```
Comment (ylorph) [inline src/EventStore.Common/Log/EventStoreLoggerConfiguration.cs:49]: the `logConfig.json` si a hidden confiugration option ?

**Review by jageall — APPROVED** [review]

**Review by ylorph — APPROVED** [review]

**Review by ylorph — CHANGES_REQUESTED** [review]
document  the logconfig.json file option

File: src/EventStore.Common/Log/EventStoreLoggerConfiguration.cs (hunk 1 above)
Comment (thefringeninja) [reply src/EventStore.Common/Log/EventStoreLoggerConfiguration.cs:49]: No it should be documented

File: docs/diagnostics/logging.md [hunk 2]
```diff
@@ -37,6 +37,16 @@ For example, adding this line to the `eventstore.conf` file will force writing l
 Log: /tmp/eventstore/logs
 ```
 
+## Log Configuration File
+
+Specifies location of file which configures the logging levels of various components.
```
Comment (ylorph) [inline docs/diagnostics/logging.md:42]:
```suggestion
Specifies the location of the file which configures the logging levels of various components.
```
- what's the default location ?
- can be full or relative ?

**Review by ylorph — DISMISSED** [review]

**Review by ylorph — APPROVED** [review]

**Review by jageall — APPROVED** [review]

**Pull request merged**